### ISSUE TITLE
fatal error: Index out of range

### ISSUE THREAD

**Issue by shenmiaoling** [yysskk/SwipeMenuViewController#7]
When return from the SwipeMenuViewController to the previous location
![2017-09-27 9 48 13](https://user-images.githubusercontent.com/17367714/30892149-9b960afe-a2fc-11e7-9e3c-b8c101dd5ef7.png)

**Comment by chsasaw** [comment]
same

**Comment by chsasaw** [comment]
@shenmiaoling 

fix by add code to ContentScrollView, SwipeMenuView, TabView:

open override func didMoveToSuperview() {
        super.didMoveToSuperview()

        if self.superview != nil {
            setup()
        }
    }

**Comment by Mirach0rus** [comment]
Same error. SwipeMenuViewController (1.1.5), Xcode 8.2.1, Swift 3.0.2

**Comment by yashdishatechnology** [comment]
same issue please help

**Comment by yysskk** [comment]
Hi, @shenmiaoling @chsasaw @Mirach0rus @yashdishatechnology .
Maybe, It happens because the number of pages and the number of tabs do not match.
I resolved this problem for fatal error crush by released [version 2.0.1](https://github.com/yysskk/SwipeMenuViewController/tree/2.0.1) .
If the number of pages and the number of tabs do not match, it will be blank instead of fatal error crush. (Xcode9.0, Swift4.x)

Thank you for reporting :-)

**Comment by yysskk** [comment]
If you want to use with Swift 3.x please fork this code and make such a fix.

- before
```
extension TabView {
    var currentItem: TabItemView {
        return itemViews[currentIndex]
    }
    ...
}
```

- after
```
extension TabView {
    var currentItem: TabItemView? {
        return currentIndex < itemViews.count ? itemViews[currentIndex] : nil
    }
    ...
}
```
